Core/include/Acts/Vertexing/NumericalTrackLinearizer.hpp

```diff
@@ -0,0 +1,164 @@
+    /// @param prop The propagator
+    Config(std::shared_ptr<const MagneticFieldProvider> bIn,
+           std::shared_ptr<const Propagator_t> prop)
+        : bField(std::move(bIn)), propagator(std::move(prop)) {}
+
+    /// @brief Config constructor without B field -> uses NullBField
+    ///
+    /// @param prop Propagator
+    Config(std::shared_ptr<const Propagator_t> prop)
+        : bField{std::make_shared<NullBField>()}, propagator(std::move(prop)) {}
+
+    std::shared_ptr<const MagneticFieldProvider> bField;
+
+    std::shared_ptr<const Propagator_t> propagator;
+
+    // Tolerance determining how close we need to get to a surface to
+    // reach it during propagation
+    ActsScalar targetTolerance = 1e-12;
+
+    // Setting size of the perturbation delta for calculation of numerical
+    // derivatives (i.e., f'(x) ~ (f(x+delta) - f(x)) / delta)
+    ActsScalar delta = 1e-8;
+  };
+
+  /// @brief Constructor
+  ///
+  /// @param config Configuration object
+  /// @param _logger Logger instance
+  NumericalTrackLinearizer(const Config& config,
+                           std::unique_ptr<const Logger> _logger =
+                               getDefaultLogger("NumTrkLinProp", Logging::INFO))
+      : m_cfg(config), m_logger{std::move(_logger)} {}
+
+  /// @brief Function that linearizes BoundTrackParameters at
+  /// the PCA to a given Perigee surface
+  ///
+  /// @param params Parameters to linearize
+  /// @param linPoint Point which defines the Perigee.
+  /// @note Transverse plane of the Perigee corresponding to @p linPoint is
+  /// parallel to the global x-y plane
+  /// @param gctx Geometry context
+  /// @param mctx Magnetic field context
+  /// @param state Linearizer state object
+  ///
+  /// @return Linearized track
+  Result<LinearizedTrack> linearizeTrack(const BoundTrackParameters& params,
+                                         const Vector4& linPoint,
+                                         const Acts::GeometryContext& gctx,
+                                         const Acts::MagneticFieldContext& mctx,
+                                         State& state) const {
+    // Using "state" to suppress unused-parameter warning
+    (void)state;
+    // Call the function without state argument
+    return linearizeTrack(params, linPoint, gctx, mctx);
+  };
+
+  /// @brief Same function as above but without the "state" argument, which we don't need for numerical track linearization.
+  /// @note We don't discard the "state" argument above since we want the templating to work (it is needed in other linearizer classes)
+  ///
+  /// @param params Parameters to linearize
+  /// @param linPoint Point which defines the Perigee.
+  /// @note Transverse plane of the Perigee corresponding to @p linPoint is
+  /// parallel to the global x-y plane
+  /// @param gctx Geometry context
+  /// @param mctx Magnetic field context
+  ///
+  /// @return Linearized track
+  Result<LinearizedTrack> linearizeTrack(
```

Comment:
Do you need a separate function for this? Is it ever called other than in the other method?